From a37a706547897d77b3194fc507b2546197def430 Mon Sep 17 00:00:00 2001
From: James Hughes <JamesH65@users.noreply.github.com>
Date: Tue, 17 Sep 2019 16:22:09 +0100
Subject: [PATCH 804/806] Fix poll rate on touchscreen (#3238)

Was running at 25Hz, rather than he expected 60. Only been doing it
for the last 5 years....

Replace msleep_interruptible with usleep_range as the msleep call
is not accurate for times < 20ms.

Fixes: https://github.com/raspberrypi/linux/issues/3227

Signed-off-by: James Hughes <james.hughes@raspberrypi.org>
---
 drivers/input/touchscreen/rpi-ft5406.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/input/touchscreen/rpi-ft5406.c
+++ b/drivers/input/touchscreen/rpi-ft5406.c
@@ -78,7 +78,7 @@ static int ft5406_thread(void *arg)
 
 	while (!kthread_should_stop()) {
 		/* 60fps polling */
-		msleep_interruptible(17);
+		usleep_range(16600, 16700);
 		memcpy_fromio(&regs, ts->ts_base, sizeof(struct ft5406_regs));
 		iowrite8(99,
 			 ts->ts_base +
